python-spec/src/somabase/data.py

```diff
@@ -100,3 +191,55 @@ def nnz(self) -> int:
 
     is_sparse: Final = True
     soma_type: Final = "SOMASparseNDArray"
+
+
+#
+# Read types
+#
+
+_T = TypeVar("_T")
+
+
+# Sparse reads are returned as an iterable structure:
+
+
+class ReadIter(Iterator[_T], metaclass=abc.ABCMeta):
+    """SparseRead result iterator allowing users to flatten the iteration."""
+
+    # __iter__ is already implemented as `return self` in Iterator
+    # SOMA implementations must implement __next__.
+
+    # XXX: Considering the name "flat" here too.
```

Comment:
`flat` or `concat` seem better to me, as they more clearly indicate what happens if the __next__ has already been called on the iterator....

Reply:
changed to `concat` (while `flat` is a bit more fun, consistency with Arrow is more useful)